### PR TITLE
Add-Power on Backend

### LINKED ISSUE #75: Allow cards to modify other cards on being played
Due to #41 we want some cards to have an effect when they are played, in particular the effect of modifying other cards.

### PR DESCRIPTION
**Pull request by Masclins**
Allows cards to have the `:add-power` attribute, modifying cards upon play (if played with a `:target`).
I thought about creating an attribute called `:abilities` and putting all its abilities inside it. Right now we only have one ability so it seemed unpractical.

This resolves #75 and outdates #102 (since I needed those commits to check the `api` without doing the work twice)

### CODE REVIEW

File: backend/test/api/effects_test.clj [hunk 1]
```diff
@@ -0,0 +1,23 @@
+(ns api.effects-test
+  (:require [expectations.clojure.test :refer :all]
+            [clojure.test :as ctest]
+            [mocking :as mocking]
+            [api.base :as api]))
+
+(ctest/use-fixtures :each mocking/mock-persistence)
+
+(defexpect playing-effects
```
Comment (Masclins) [inline backend/test/api/effects_test.clj]: I'm not sure this test is needed, but better to be safe than sorry.

File: backend/src/rules/play_card.clj [hunk 2]
```diff
@@ -23,14 +24,26 @@
...
+  (if (nil? (:target play))
+    game-state
+    (alter-card/add-power
+      game-state 
+      (:target play)
+      (get-in game-state [:players (:player play)
+                          :hand (:index play)
+                          :add-power]))))
+
+(defn apply-play-card
```
Comment (Masclins) [inline backend/src/rules/play_card.clj]: No longer private so we can call it from tests and make them more targeted to what we wanna test.

**Comment by Masclins** [comment]
I thought about another alternative, but I feel like it doesn't have much sense now that we only have one ability:

- *cards* have `:abilities {:add-power x :destroy-card true}` (`true` or `nil` would be used for abilities that require no value. Also could be used for adding future restrictions there)

- *next-play* have `:targets {:add-power target :ability-n2 target}`

File: backend/test/rules/effects_test.clj [hunk 3]
```diff
@@ -33,3 +34,12 @@
       [nil {:croissant "chocolate" :enchilada {:power 1 :salsa "mild"}}]
       [1 :enchilada]
       55)))
+
+(defexpect playing-add-power
```
Comment (kenan-rhoton) [inline backend/test/rules/effects_test.clj]: Would like a second test with different values here 🙃

File: backend/test/api/effects_test.clj (hunk 1 above)
Comment (kenan-rhoton) [reply backend/test/api/effects_test.clj]: The test is fine for now, although it's the kind of test I'm wanting to simplify with #98

**Pull request closed without merging**